docs/customization/extensions.rst

```diff
@@ -0,0 +1,75 @@
+Extensions
+==========
+Saleor has implemented extensions architecture.
+It includes hooks for most basic operations like calculation of prices in the checkout or
+calling some actions when an order has been created.
+
+
+ExtensionsManager
+-----------------
+ExtensionsManager is located in ``saleor.core.extensions.base_plugin``.
+It is a class responsible for handling all declared plugins and serving a response from them.
+It serves a default response in case of a non-declared plugin.  There is a possibility to overwrite ExtensionsManager
+class by implementing it on its own. Saleor will discover the manager class by taking declared path from
+``settings.EXTENSIONS_MANAGER``.
+Each Django request object has its own manager included as the ``extensions`` field. It is attached in the Saleor middleware.
+
+
+BasePlugin
+----------
+BasePlugin is located in ``saleor.core.extensions.base_plugin``. It is an abstract class for storing all methods
+available for any plugin. All methods take the ``previous_value`` parameter. ``previous_value`` contains a value
+calculated by the previous plugin in the queue. If the plugin is first, it will use the default value calculated by
+the manager.
+
+
+Plugins
+-------
+Saleor has some plugins implemented by default. These plugins are located in ``saleor.core.extensions.plugins``.
+The ExtensionManager needs to receive a list of enabled plugins. It can be done by including the plugin python path in the
```

Comment:
Maybe we should provide an example of how it might look e.g. something like:
```
PLUGINS = ["saleor.core.plugins.maciek_plugin"]
```
Might be helpful. Or do we have already some enabled by default?